### DISCUSSION TITLE
SOLVED: Should non-integer component multiplications be allowed?

### DISCUSSION THREAD

**Discussion by keckler** [terrapower/armi#542] [General]
It appears that a user can enter floats for the `mult` attribute when constructing components, and ARMI does not complain. 

Should we allow this? 

On the one hand, possibly somebody could exploit this as a modeling trick. One the other, it is unexpected and (probably) non-physical, and adding a type check with `yamlize` to ensure it is an `int` could help with reducing input errors.

**Comment by sombrereau** [comment]
yes, or some sort of support for this. I exploit this as a last ditch effort to make models of 3D shapes that can not be defined using ARMI's existing component types.

**Reply by keckler** [reply]
Good to know, thanks for chiming in. Are you just adjusting the component multiplication until you get the desired volume/area for a component, or something similar? Or are you actually trying to represent fractions of the component shapes, like using 0.5 * circle to make a semi-circle? Just curious.

**Reply by john-science** [reply]
This is interesting, Tommy.  I'd be curious to know more about why this is useful to you.

Before you said anything I thought this was a flat "well, no, that would be crazy".

At the very least, I think we need a big warning for non-integer values. But I'd still be interested to hear what Tommy thinks.

**Reply by john-science** [reply]
@sombrereau 

Do you know about `UnshapedComponent`?

https://github.com/terrapower/armi/blob/e2a064c7c2c412bad26ae8a39ba0a2544c8bcd34/armi/reactor/components/__init__.py#L116-L121

Or `UnshapedVolumetricComponent`?

https://github.com/terrapower/armi/blob/e2a064c7c2c412bad26ae8a39ba0a2544c8bcd34/armi/reactor/components/__init__.py#L192-L197

I guess I have two questions:

1. Do you think these would work for your needs?
2. If not, what could I provide to you that _would_?

Because, on balance, I like the idea of making the `mult` an integer. Though I may not understand user's needs well enough; that's always possible.

**Comment by john-science** [comment]
Okay, we have an answer.

Nick (@ntouran) says it was always a design decision to explicitly allow a float in the `mult` parameter. In case people found it useful. And it seems like some people already have.

* Maybe someone wants 3/4 of a circle.
* Maybe someone wants a cylinder sliced in half vertically.

So, yes, this is a feature. We live and learn.

**Reply by keckler** [reply]
I guess, though, this does not preclude that a warning might be useful regardless? Yes, it is a feature. But it is kind of an unexpected feature, perhaps?
